pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -553,6 +554,16 @@ private void createDefaultValueColumnV1Indices(String column)
       }
     }
 
+    if (!_segmentWriter.hasIndexFor(column, ColumnIndexType.NULLVALUE_VECTOR)) {
```

Comment:
Should the `nullValueVector` be created only when `nullHandlingEnabled` is set to true?

Reply:
Fixed now.